shell/browser/api/electron_api_web_request.cc

```diff
@@ -185,37 +185,40 @@ void FillDetails(gin_helper::Dictionary* details, Arg arg, Args... args) {
   FillDetails(details, args...);
 }
 
-// Fill the native types with the result from the response object.
-void ReadFromResponse(v8::Isolate* isolate,
-                      gin::Dictionary* response,
-                      GURL* new_location) {
-  response->Get("redirectURL", new_location);
-}
-
-void ReadFromResponse(v8::Isolate* isolate,
-                      gin::Dictionary* response,
-                      net::HttpRequestHeaders* headers) {
-  v8::Local<v8::Value> value;
-  if (response->Get("requestHeaders", &value) && value->IsObject()) {
-    headers->Clear();
-    gin::Converter<net::HttpRequestHeaders>::FromV8(isolate, value, headers);
-  }
-}
+std::pair<std::set<std::string>, std::set<std::string>>
+CalculateOnBeforeSendHeadersDelta(net::HttpRequestHeaders* old_headers,
+                                  net::HttpRequestHeaders* new_headers) {
+  // Newly introduced or overridden request headers.
+  std::set<std::string> modified_request_headers;
+  // Keys of request headers to be deleted.
+  std::set<std::string> deleted_request_headers;
+
+  // The event listener might not have passed any new headers if it
+  // just wanted to cancel the request.
+  if (new_headers) {
```

Comment:
Upstream passes two raw pointers: https://source.chromium.org/chromium/chromium/src/+/main:extensions/browser/api/web_request/web_request_api_helpers.h;l=387-388;bpv=1;bpt=1?q=CalculateOnBeforeSendHeadersDelt&sq=package:chromium&type=cs

- this check is lightly modified from there: https://source.chromium.org/chromium/chromium/src/+/main:extensions/browser/api/web_request/web_request_api_helpers.cc;l=792-794;drc=ae983e94583c5e2d400fa7641439d2e8ca13fd93;bpv=1;bpt=1

I think our previous code might be incorrect and old_headers will always be present, since they deference that without checking in the same helper 🤔